server/internal/usecase/interfaces/dataset.go

```diff
@@ -62,6 +63,7 @@ var (
 
 type Dataset interface {
 	Fetch(context.Context, []id.DatasetID, *usecase.Operator) (dataset.List, error)
+	Export(context.Context, id.DatasetSchemaID, *usecase.Operator) (io.Reader, string, error)
```

Comment:
```suggestion
	Export(context.Context, id.DatasetSchemaID, string, *usecase.Operator) (io.Reader, string, error)
```

The name "Export" does not include the intention to export CSV, but in fact it does; why not name it ExportCSV or allow the format to be specified as an argument?